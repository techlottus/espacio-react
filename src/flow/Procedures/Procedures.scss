@import "../../App.scss";

.procedures-container {
  width: 100%;
  @extend .headerapp-top;
}
