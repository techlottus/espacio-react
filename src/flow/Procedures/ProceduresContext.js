import { createContext } from "react";

export const ProceduresContext= createContext(null) 