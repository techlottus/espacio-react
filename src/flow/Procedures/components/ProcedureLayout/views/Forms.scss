@import "../../../../../App.scss";

.forms-container {
  font-family: "Nunito",sans-serif;
  padding: pixeltorem(38px) pixeltorem(25px) 0 pixeltorem(25px);
  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    padding: pixeltorem(38px) pixeltorem(35px) 0 pixeltorem(114px);
    max-width: 50%;
  }
  .forms-view {
    background-color: map-get($palette-neutral, n00);

    .forms-stepper {
      height: pixeltorem(100px);
      width: 100%;
      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        width: 100%;
        padding: 0 0;
      }
    }
  }
}
