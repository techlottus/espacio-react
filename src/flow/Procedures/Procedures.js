import React, { useEffect, useState } from "react";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import "./Procedures.scss";
import { ProceduresContext } from "./ProceduresContext";
import {
  stepsProcedures,
  typeflowsProcedures,
} from "../../constants/Procedures.constant";
import ProcedureLayout from "./components/ProcedureLayout/ProcedureLayout";
import { ModalProcedures } from "../../components/Modal/ModalProcedures/ModalProcedures";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { proceduresRequest } from "../../middlewares/proceduresMiddlewares/indexProceduresMiddleware";
import { getProceduresCostsService } from "../../middlewares/proceduresMiddlewares/proceduresCostsMiddleware";
import { ModalAccordion } from "../../components/Modal/ModalAccordion/ModalAccordion";
import { getProfileProceduresService } from "../../middlewares/proceduresMiddlewares/procedureMainMiddleware";
import { ModalExit } from "../../components/Modal/ModalExit/ModalExit";
import { setPathExitProcedures } from "../../actions/proceduresActions/proceduresAction";
import { unblockProcedures } from "../../helpers/unblock";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import {
  modalProcedureBackEquivalenceStudiesTag,
  nextStepTwoEquivalenceStudiesTag,
} from "../../tagging/flows/procedures/equivalenceStudiesTag";
import {
  modalProcedureBackAcademicHistoryTag,
  modalProcedureNextAcademicHistoryTag,
} from "../../tagging/flows/procedures/academicHistoryTag";
import { certificateStudyVerifyTag } from "../../tagging/flows/procedures/certificateStudyTag";
import {
  modalVerifyBackStudyRecordTag,
  modalVerifyNextStudyRecordTag,
} from "../../tagging/flows/procedures/studyRecordTag";
import {
  modalVerifyBackReinstatementTag,
  modalVerifyNextReinstatementTag,
} from "../../tagging/flows/procedures/reinstatementTag";
import { ModalRequirementsProcedure } from "../../components/Modal/ModalRequirementsProcedure/ModalRequirementsProcedure";
import { ModalImage } from "../../components/Modal/ModalImage/ModalImage";

const Procedures = () => {
  const [steps, setSteps] = useState({
    ...stepsProcedures,
  });

  const history = useHistory();
  const [isModal, setIsModal] = useState({
    valid: false,
    info: null,
    file: null,
    type: null,
  });

  const [isModalExit, setIsModalExit] = useState({
    isShow: false,
    unblock: null,
  });

  const [isModalAcordeon, setIsModalAcordeon] = useState(false);
  const [isModalRequirements, setIsModalRequirements] = useState({
    isShow: false,
    info: []
  });
  const [isModalImage, setIsModalImage] = useState(false);
  const [isTabs,setIsTabs] = useState(true);
  const dispatch = useDispatch();
  const { proceduresAllOne,proceduresAllTwo } = useSelector((state) => state.procedures);
  const procedures = useSelector((state) => state.procedures);

  const handleStepsBody = (focus, titles ) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
        titles
      };
    });
  };

  const handleSteps = (focus) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
      };
    });
  };

  const handleAccordion = (state) => {
    setIsModalAcordeon(state);
  };

  const handleIsTabs = (isCheck) => {
    setIsTabs(isCheck);
  }

  const handleModalRequirements = (isShow,info) => {
    setIsModalRequirements({isShow,info});
  };

  const handleModalImage = (state) => {
    setIsModalImage(state);
  };

  const handleModalProcedures = (info) => {
    setIsModal(info);
  };
  const handleRequest = () => {
    setIsModal(false);
    dispatch(
      proceduresRequest(
        isModal.type,
        history,
        isModal.info,
        isModal.file,
        isModal.detailId
      )
    );
  };

  const [infoModal, setInfoModal] = useState({
    profile: null,
    information: null,
    file: null,
    title: null,
    cost: null,
  });
  const handleInfoModalProcedures = (info) => {
    setInfoModal(info);
  };
  useEffect(() => {
    let urlSplit = history.location.pathname.split("/");

    let procedureFlow = urlSplit[2];
    let unblock = history.block((location) => {
      if (unblockProcedures(location.pathname, procedureFlow)) {
        dispatch(setPathExitProcedures(location.pathname));
        setIsModalExit({
          isShow: true,
          unblock,
        });
        return false;
      }

      return true;
    });

    if (proceduresAllOne === null || proceduresAllTwo === null) {
      dispatch(getProfileProceduresService());
      dispatch(getProceduresCostsService());
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <div className="procedures-container">
        <HeaderApp />
        <ProceduresContext.Provider
          value={{
            steps,
            handleStepsBody,
            handleSteps,
            isModal,
            isTabs,
            handleModalProcedures,
            handleInfoModalProcedures,
            handleAccordion,
            handleModalRequirements,
            handleIsTabs,
            handleModalImage
          }}
        >
          <ProcedureLayout />
          <ModalExit
            isShow={isModalExit.isShow}
            onClose={() => {
              setIsModalExit((state) => {
                return {
                  ...state,
                  isShow: false,
                };
              });
            }}
            onNext={() => {
              isModalExit.unblock();
              setIsModalExit({
                isShow: false,
                unblock: null,
              });
              setPathExitProcedures(null);
              history.push(procedures.pathExit);
            }}
            flow={"procedures"}
          />
        </ProceduresContext.Provider>
        <ModalProcedures
          isShow={isModal.valid}
          outside={() => setIsModal(false)}
          onClose={() => {
            setIsModal(false);
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "back");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, modalProcedureBackEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureBackAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyBackStudyRecordTag, "back");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyBackReinstatementTag, "back");
                break;
              default:
                break;
            }
          }}
          onNext={() => {
            handleRequest();
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "next");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, nextStepTwoEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureNextAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyNextStudyRecordTag, "next");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyNextReinstatementTag, "next");
                break;
              default:
                break;
            }
          }}
          infoModal={infoModal}
        />
        <ModalAccordion
          isShow={isModalAcordeon}
          onClose={() => {
            handleAccordion(false);
          }}
          outside={() => handleAccordion(false)}
        />
        <ModalRequirementsProcedure
          isShow={isModalRequirements.isShow}
          info={isModalRequirements.info}
          onClose={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
          outside={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
        />
        <ModalImage
          isShow={isModalImage}
          onClose={() => {
            handleModalImage(false);
          }}
          outside={() => handleModalImage(false)}
        />
      </div>
    </>
  );
};

export default Procedures;
