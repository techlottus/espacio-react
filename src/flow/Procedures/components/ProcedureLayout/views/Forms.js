import React, { useContext } from "react";
import { Stepper } from "../../../../../components/Stepper/Stepper";
import ProceduresRoutes from "../../../../../routes/ProceduresRoutes";
import { ProceduresContext } from "../../../ProceduresContext";

import "./Forms.scss";

const Forms = () => {
  const { steps, isTabs } = useContext(ProceduresContext);

  return (
    <>
      <div className="forms-container">
        <div className="forms-view">
          {isTabs ? (
            <div className="forms-stepper">
              <Stepper data={steps} />
            </div>
          ) : (
            ""
          )}
          <div className="forms-content">
            <ProceduresRoutes />
          </div>
        </div>
      </div>
    </>
  );
};

export default Forms;
