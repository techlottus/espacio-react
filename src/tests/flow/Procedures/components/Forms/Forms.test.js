import React from "react";
import { mount } from "enzyme";
import Forms from "../../../../../flow/Procedures/components/ProcedureLayout/views/Forms";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ProceduresContext } from "../../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {};

const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
// const type = jest.fn().mockImplementation({ flow: "current" });
const steps = {
  focus: 0,
  titles: [
    {
      text: "Información",
      disabled: false,
    },
    {
      text: "Documentación",
      disabled: false,
    },
  ],
  off: false,
};
let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{steps}}>
      <Provider store={store}>
        <BrowserRouter>
          <Forms />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <Forms/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
