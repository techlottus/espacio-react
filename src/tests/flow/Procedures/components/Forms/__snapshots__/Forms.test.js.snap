// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Forms/> should init component 1`] = `
<div
  className="forms-container"
>
  <div
    className="forms-view"
  >
    <div
      className="forms-stepper"
    >
      <lottus-stepper />
    </div>
    <div
      className="forms-content"
    />
  </div>
</div>
`;
